### ISSUE TITLE
Merlin hangs

### ISSUE THREAD

**Issue by mattam82** [ocaml/merlin#929]
I just found that merlin hangs while editing, the process uses 100% cpu suddenly when editing 
the src/covering.ml file in this branch https://github.com/mattam82/Coq-Equations/commits/merlin-bug (which compiles again Coq's master commit ff2c1bea58fcb53c09ba58c4fd3f4d96e750fcbd). Specifically when editing line around 90. This is using merlin 3.2.2. Not sure if you can reproduce that, but I need to kill -9 ocamlmerlin-server from time to time.

**Comment by let-def** [comment]
I cannot reproduce the bug.
Can you share the output of `M-x merlin-debug-last-commands`?

**Comment by mattam82** [comment]
I found the culprit a bit after, there was a “syntax” error above, a missed closing parenthesis, that was hard to detect as an error I suppose. Could that explain it? Is merlin supposed to be robust in case it gets large nonsensical terms? I’ll try to reproduce next week otherwise

**Comment by let-def** [comment]
Yes, the parser should be robust (in theory, it should terminate on any input, although the implementation deviates slightly from the theory, so if that's the case that would be very interesting to investigate). I got tuareg's parser stuck on some input but not merlin's one.
If you manage to get it stuck again, could you attach gdb to the process and provide a backtrace of ocamlmerlin-server?
